connectors/sources/notion.py

```diff
@@ -0,0 +1,462 @@
+                    self._get_client.blocks.children.list, block_id=block.get("id")
+                ):
+                    yield child_block
+
+        async for block in async_iterate_paginated_api(
+            self._get_client.blocks.children.list, block_id=block_id
+        ):
+            if block.get("type") not in ["child_database", "child_page"]:
+                yield block
+                async for child in fetch_children_recursively(block):
+                    yield child
+            if block.get("type") == "child_database":
+                async for record in self.query_database(block.get("id")):
+                    yield record
+
+    async def fetch_by_query(self, query):
+        async for document in async_iterate_paginated_api(
+            self._get_client.search, **query
+        ):
+            yield document
+            if query and query.get("filter", {}).get("value") == "database":
+                async for database in self.query_database(document.get("id")):
+                    yield database
+
+    async def fetch_comments(self, block_id):
+        async for block_comment in async_iterate_paginated_api(
+            self._get_client.comments.list, block_id=block_id
+        ):
+            yield block_comment
+
+    async def query_database(self, database_id):
+        async for result in async_iterate_paginated_api(
+            self._get_client.databases.query, database_id=database_id
+        ):
+            yield result
+
+
+class NotionDataSource(BaseDataSource):
+    """Notion"""
+
+    name = "Notion"
+    service_type = "notion"
+
+    def __init__(self, configuration):
+        """Setup the connection to the Notion instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.notion_client = NotionClient(configuration=configuration)
+
+        self.configuration = configuration
+        self.index_comments = self.configuration["index_comments"]
+        self.pages = self.configuration["pages"]
+        self.databases = self.configuration["databases"]
+        self._logger = logger
+        self._sleeps = CancellableSleeps()
+        self.concurrent_downloads = self.configuration["concurrent_downloads"]
+
+    def _set_internal_logger(self):
+        self.notion_client.set_logger(self._logger)
+
+    async def ping(self):
+        """Verify the connection with Notion"""
+        try:
+            await self.notion_client.fetch_owner()
+            self._logger.info("Successfully connected to Notion.")
+        except Exception:
+            self._logger.exception("Error while connecting to Notion.")
+            raise
+
+    async def close(self):
+        """Closes unclosed client"""
+        await self.notion_client.close()
+
+    async def _remote_validation(self, entity_type, entity_titles):
```

Comment:
I think the name is a bit misleading here, you look for specific entity types and titles, right? Maybe it should be called something like `get_entities` or something similar?